craftai/pandas/utils.py

```diff
@@ -161,3 +164,91 @@ def display_tree(tree_object, decision_path="",
                  height=500):
   tree_html = create_tree_html(tree_object, decision_path, edge_type, folded_nodes, height)
   display(HTML(tree_html))
+
+def _flatten(arr, flat=None):
+  """ flatten deep an array of lists """
+  if flat is None:
+    flat = []
+  for value in arr:
+    if not isinstance(value, list) and value not in flat:
+      flat.append(value)
+  if sum([isinstance(value, list) for value in arr]) > 0:
+    return _flatten([x for y in arr for x in y if isinstance(y, list)], deepcopy(flat))
+  return deepcopy(flat)
+
+def _update_paths(paths, idx):
+  """ add new path build on idx to all paths """
+  if paths:
+    return paths + ["{}-{}".format(paths[-1], idx)]
+  return [str(idx)]
+
+def _paths(tree, paths=None):
+  """ return a raw list of all paths in a tree """
+  if paths is None:
+    paths = ["0"]
+  if "children" in tree.keys():
```

Comment:
```suggestion
  if "children" in tree:
```
https://docs.python.org/3/library/stdtypes.html#dict

Reply:
I didn't know this syntax, thx :+1: